test/parallel/test-tz-version.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasIntl) {
+  common.skip('missing Intl');
+}
+
+// Refs: https://github.com/nodejs/node/blob/1af63a90ca3a59ca05b3a12ad7dbea04008db7d9/configure.py#L1694-L1711
+if (process.config.variables.icu_path !== 'deps/icu-small') {
+  common.skip('not using the icu data file present in deps/icu-small/source/data/in/icudt##l.dat.bz2');
+}
```

Comment:
So this would work only if `node` is built with `--with-intl=small-icu`? Shouldn't we also check that if the value is `undefined`, we want to also run the test?

Reply:
> So this would work only if `node` is built with `--with-intl=small-icu`?

Looks like that's not true:
https://github.com/nodejs/node/blob/7124476fb3a9a58fd316f9cef3329c14fc1cd966/configure.py#L1703-L1711
i.e., it should work if `--with-intl` is set to `full-icu` (default case).

> Shouldn't we also check that if the value is `undefined`, we want to also run the test?

I don't think that's necessary because `icu_path` can never be `undefined` for the cases we are interested in (the objective is to run the test iff the current build is using `deps/icu-small/source/data/in/icudt##l.dat.bz2`):
* `icu_path` is set here and `icu_full_path` is never undefined - https://github.com/nodejs/node/blob/7124476fb3a9a58fd316f9cef3329c14fc1cd966/configure.py#L1761
* And this line wouldn't run only if we hit a `return` statement between https://github.com/nodejs/node/blob/7124476fb3a9a58fd316f9cef3329c14fc1cd966/configure.py#L1584 and this assignment. The possible returns are:
  * https://github.com/nodejs/node/blob/7124476fb3a9a58fd316f9cef3329c14fc1cd966/configure.py#L1641 - this would happen only if we pass `--with-icu-path`, I'm avoiding this to exclude the case where the pointed gyp file attempts to build icu with a different icudt file, which would result in a different `process.versions.tz` value
  * https://github.com/nodejs/node/blob/7124476fb3a9a58fd316f9cef3329c14fc1cd966/configure.py#L1646 - this would happen only if we pass `--with-intl=none` or `--without-intl`, which is not being considered because `process.versions.tz` would be `undefined` in this case
  * https://github.com/nodejs/node/blob/7124476fb3a9a58fd316f9cef3329c14fc1cd966/configure.py#L1682 - this would happen only if we pass `--with-intl=system-icu` which we are avoiding because the system icu is not in our control, so it might produce a different version for `process.versions.tz`

Reply:
On macOS, `node` installed through `brew`:

```console
$ node -p '"Intl" in globalThis'
true
$ node -p 'process.config.variables.icu_path'            
undefined
```

I tested this with Node.js v14.x, v16.x, and v18.x.

Reply:
> Maybe it's using system-icu though,

That is correct.

> I don't know if there's another way to check.

There is :)

```node
> process.config.variables.icu_gyp_path
'tools/icu/icu-system.gyp'
```

It relies on the gyp file for the systemwide icu instead of `tools/icu/icu-generic.gyp` which is for the builtin icu.
Alternatively, `otool -L /usr/local/opt/node/bin/node` would print a set of shared objects the binary dynamically links against and you'll see the icu dylibs in that list:
```
	/usr/local/opt/icu4c/lib/libicui18n.71.dylib (compatibility version 71.0.0, current version 71.1.0)
	/usr/local/opt/icu4c/lib/libicuuc.71.dylib (compatibility version 71.0.0, current version 71.1.0)
	/usr/local/opt/icu4c/lib/libicudata.71.dylib (compatibility version 71.0.0, current version 71.1.0)
```

> In any case, this should probably be explained in a comment why checking for 'deps/icu-small' covers more than small ICU builds.

Explained, PTAL.